File: backbrace/lambda/python/handler.py
import json
from typing import Any

from python import python


def handler(event: Any, context: Any) -> dict[Any, Any]:  # type: ignore
    path = event.get("path", "")

    try:
        if path == "/python":
            return python(event, context)

        return {
            "statusCode": 404,
            "body": json.dumps({"message": "path not found", "path": path}),
        }

    except Exception as e:
        print("Error:", str(e))
        return {
            "statusCode": 500,
            "body": json.dumps({"message": "Internal server error", "error": str(e)}),
        }

File: backbrace/lambda/javascript/handler.ts
import { bedrock } from "./bedrock";
import { send } from "./res";

export const handler = async (event: any, context: any) => {
  try {
    console.log("Full event:", JSON.stringify(event, null, 2));
    const path =
      event.resource || event.requestContext?.resourcePath || event.path;
    console.log("Path:", path); // mess w cdk stack incoming to get path

    return bedrock(event, context);

    // switch (path) {
    //   case "/bedrock":
    //     return bedrock(event, context);
    //   default:
    //     return send(404, { message: "Not Found" });
    // }
  } catch (error) {
    console.error("Error processing request:", error);
    return send(500, {
      message: "Internal Server Error",
      details: (error as any).message,
    });
  }
};
